frameworks/torch/torch-neuronx/tutorials/training/bert.rst

```diff
@@ -329,6 +330,47 @@ the BERT pretraining demo:
+-----------------------------------------------
+Sometimes, to reduce the training wall time, you can use higher learning rate and larger global batch size. The apporach is discussed in `LARGE BATCH OPTIMIZATION FOR DEEP LEARNING: TRAINING BERT IN 76 MINUTES <https://arxiv.org/pdf/1904.00962.pdf>`__. Tranium supports LAMB, and in this tutorial, we use a public available XLA-friendly LAMB implemenation from https://github.com/rwightman/pytorch-image-models/blob/master/timm/optim/lamb.py. You need to download the lamb script before running the tutorial. 
+
+.. code:: bash
+
+   cd ~/examples/dp_bert_hf_pretrain
+   wget https://raw.githubusercontent.com/rwightman/pytorch-image-models/main/timm/optim/lamb.py .
+   torchrun --nproc_per_node=32 \
+   dp_bert_large_hf_pretrain_hdf5.py \
+   --batch_size 8 \
+   --optimizer LAMB \
+   --lr 6e-3 \
+   --grad_accum_usteps 256 |& tee run_pretrain_log.txt
+ 
+The command-line argument ``--optimizer LAMB`` is needed, otherwise, the default optimizer AdamW will be used. Besides, you need to use a set of hyper-parameters supporting the larger global batch size (GBS). In this case, we have 64k as GBS for LAMB and use a set of hyper-params similar to https://github.com/NVIDIA/DeepLearningExamples/blob/master/PyTorch/LanguageModeling/BERT/README.md. Given higher GBS from LAMB than AdamW, it takes fewer steps (roughly 8k) to achieve similar level of accuracy as AdamW, which takes more than 28k steps. In addition, you can also use different data types on top of LAMB. Below is an example using the BFloat16 and Stochastic Roundings. 
+
+.. code:: bash
+
+   cd ~/examples/dp_bert_hf_pretrain
+   wget https://raw.githubusercontent.com/rwightman/pytorch-image-models/main/timm/optim/lamb.py .
+   XLA_DOWNCAST_BF16=1 torchrun --nproc_per_node=32 \
+   dp_bert_large_hf_pretrain_hdf5.py \
+   --batch_size 16 \
+   --optimizer LAMB \
+   --lr 6e-3 \
+   --grad_accum_usteps 128 |& tee run_pretrain_log.txt
+   
+Phase 1 BERT-Large pretraining with Automatic Mxed Precison (AMP)
+-----------------------------------------------
+Besides the `BFloat16 and stochastic rounding in phase 1 <https://github.com/aws-neuron/aws-neuron-sdk/edit/lamb-and-amp-tutorial-update/frameworks/torch/torch-neuronx/tutorials/training/bert.rst#id33>`__, you can also use AMP. The detailed background is at https://pytorch.org/docs/stable/amp.html. It uses both data type BFloat16 and Float32. With this setting, the loss ususally matches to Float32, but with a higher throughput. In applications sensitive to accuracy/loss values, AMP may be prefered over BFloat16 and Stochastic Roundings. A detailed comparison is available at `benchmark <https://awsdocs-neuron.readthedocs-hosted.com/en/latest/general/benchmarks/trn1/trn1-performance.html?highlight=performance#training-performance>`__.
+To launch the AMP, one additional command-line argument is needed ``--enable_pt_autocast``.
+
+.. code:: bash
+
+   cd ~/examples/dp_bert_hf_pretrain
```

Comment:
Please add 

NEURON_STOCHASTIC_ROUNDING_EN=1

Reply:
Thanks. added.

Reply:
Sorry Jie, I meant add it to the AMP script NEURON_RT_STOCHASTIC_ROUNDING_EN=1. 

Also, I gave the wrong envvar name. Should have "_RT" in the name, like NEURON_RT_STOCHASTIC_ROUNDING_EN.

I don't know if it is needed for LAMP.

Reply:
okay, I think once the BF16 flag is used, NEURON_RT_STOCHASTIC_ROUNDING_EN will be turn on automatically.